app/models/proxy.rb

```diff
@@ -584,23 +587,19 @@ def policies_normalize(parsed_policies)
     end
 
     def policies_configs_are_correct
-      # TODO: 5: errors.merge!(policy_config.errors)
-      select(&:invalid?).map(&:errors).each do |policy_errors|
-        policy_errors.each do |attribute, message|
-          errors.add(attribute, errors.full_message(attribute, message).downcase)
-        end
-      end
-      errors.add(:base, :missing_apicast) unless detect(&:default?)
+      errors.add(:policies_config, :invalid_policy) if any?(&:invalid?)
+      errors.add(:policies_config, :missing_apicast) unless detect(&:default?)
+      errors.add(:policies_config, :too_long, count: MAX_LENGTH) if to_s.size > MAX_LENGTH
     rescue NoMethodError
-      errors.add(:base, :invalid_format)
+      errors.add(:policies_config, :invalid_format)
     end
   end
 
   def policies_config_structure
     return if policies_config.valid?
 
-    policies_config.errors.each do |attribute, message|
-      errors.add(:policies_config, errors.full_message(attribute, message))
+    policies_config.errors.each do |_attribute, message|
+      errors.add(:policies_config, message)
```

Comment:
From `Proxy` point of view, there are in fact three error levels:

- Proxy errors
- Policy chain errors
- Policy errors

We already have policy errors promoted to chain errors, we also need to promote these to proxy errors for rails to detect them and return `422`.

This error promotion was generating wrong error messages like `'Policies config Policies config is invalid'`, adding the field name twice.

Reply:
I think I have seen this but apparently didn't check the cause of it. Good catch!